packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -558,3 +558,71 @@ class ValueNotifier<T> extends ChangeNotifier implements ValueListenable<T> {
   @override
   String toString() => '${describeIdentity(this)}($value)';
 }
+
+/// A [ValueListenable] that computes its value from other listeners.
+///
+/// This notifier listens to a list of other listeners and calls a compute function
+/// whenever any of them change to determine its own value.
+///
+/// The listeners are merged into a single listenable using [Listenable.merge].
+///
+/// ## Example
+///
+/// ```dart
+/// final ValueNotifier<int> a = ValueNotifier<int>(1);
+/// final ValueNotifier<int> b = ValueNotifier<int>(2);
+///
+/// final ComputedNotifier<int> sum = ComputedNotifier<int>(
+///   [a, b, c, d],
+///   () => a.value + b.value,
```

Comment:
What worries me about this API design: It's gonna be very easy to use a listenable in the compute function and accidentally forget to include it in the listenable list...

Reply:
You're absolutely right!! But although I think it might happen, I am not very worried about that.. It's a basic programming skill issue, not very related to the API signature.
Also, you might not want to include all values ​​used in the computed callback in your listenable list, right?

But, what do you think we can do to improve it? Besides a clear documentation..

Reply:
Maybe we can return the listenable list as a parameter in our callback, so it can reinforces the user to add the listenable to the list. 
But it creates a typing issue, once the `_listenableList` has a dynamic subtype: `List<ValueListenable<dynamic>>`. 

See the trade-off of my point:

_With typed list by generics_
```dart
final ComputedNotifier<String, int> sum = ComputedNotifier<int>(
  <ValueListenable<int>>[a, b],
  (List<ValueListenable<int>> list) => (list.first.value + list[1].value).toString(),
);
```

_with dynamic list_
```dart
final ComputedNotifier<String> sum = ComputedNotifier<int>(
  <ValueListenable<int>>[a, b],
  (List<ValueListenable<dynamic>> list) => ((list.first.value as int) + (list[1].value as int)).toString(),
);
```

_**To be honest, I do not found this a good solution. Maybe because I don't see it as a ruge problem.**_

Reply:
You made me intriguer about a solution for this... what do you think about write the API as this:

```dart
  final ValueNotifier<int> a = ValueNotifier<int>(1);
  final ValueNotifier<int> b = ValueNotifier<int>(2);

  final ComputedNotifier<int> sum = ComputedNotifier<int>(
      (ValueGetter getter) => getter(a) + getter(b),
  );

  int sumValue = sum.value;
  sum.addListener(() => sumValue = sum.value);

  a.value = 2;
  print(sumValue); // 4

  b.value = 3;
  print(sumValue); // 5
```